[Flaky test] Gang scheduling

Seeing the test failure in https://github.com/kubernetes-sigs/ai-conformance/pull/83, see https://prow.k8s.io/view/gs/kubernetes-ci-logs/pr-logs/pull/kubernetes-sigs_ai-conformance/83/pull-ai-conformance-e2e-gcp/2089823305400848384:

```
{"Time":"2026-08-18T21:38:47.801590201Z","Action":"output","Package":"github.com/kubernetes-sigs/ai-conformance/test","Test":"TestGangScheduling/NegativeGangScheduling","Output":"    gang_scheduling_test.go:244: Failed to verify negative job: client rate limiter Wait returned an error: rate: Wait(n=1) would exceed context deadline\n"}
{"Time":"2026-08-18T21:38:47.80772141Z","Action":"output","Package":"github.com/kubernetes-sigs/ai-conformance/test","Test":"TestGangScheduling","Output":"--- FAIL: TestGangScheduling (42.06s)\n"}
{"Time":"2026-08-18T21:38:47.80775572Z","Action":"output","Package":"github.com/kubernetes-sigs/ai-conformance/test","Test":"TestGangScheduling/PositiveGangScheduling","Output":"    --- PASS: TestGangScheduling/PositiveGangScheduling (12.03s)\n"}
{"Time":"2026-08-18T21:38:47.80776482Z","Action":"pass","Package":"github.com/kubernetes-sigs/ai-conformance/test","Test":"TestGangScheduling/PositiveGangScheduling","Elapsed":12.03}
{"Time":"2026-08-18T21:38:47.807776305Z","Action":"output","Package":"github.com/kubernetes-sigs/ai-conformance/test","Test":"TestGangScheduling/NegativeGangScheduling","Output":"    --- FAIL: TestGangScheduling/NegativeGangScheduling (30.02s)\n"}
{"Time":"2026-08-18T21:38:47.807781158Z","Action":"fail","Package":"github.com/kubernetes-sigs/ai-conformance/test","Test":"TestGangScheduling/NegativeGangScheduling","Elapsed":30.02}
{"Time":"2026-08-18T21:38:47.807786891Z","Action":"fail","Package":"github.com/kubernetes-sigs/ai-conformance/test","Test":"TestGangScheduling","Elapsed":42.06}
{"Time":"2026-08-18T21:38:47.807795927Z","Action":"output","Package":"github.com/kubernetes-sigs/ai-conformance/test","Output":"FAIL\n"}
{"Time":"2026-08-18T21:38:47.814122821Z","Action":"output","Package":"github.com/kubernetes-sigs/ai-conformance/test","Output":"FAIL\tgithub.com/kubernetes-sigs/ai-conformance/test\t154.782s\n"}
{"Time":"2026-08-18T21:38:47.814156891Z","Action":"fail","Package":"github.com/kubernetes-sigs/ai-conformance/test","Elapsed":154.783}
================================================================
Cleaning up E2E test environment (Exit Code: 1)
================================================================
```

This test is potentially flaky. cc @krrish175-byte

--- a/test/gang_scheduling_test.go
+++ b/test/gang_scheduling_test.go
@@ -238,7 +238,7 @@ func verifyJobSuspendedOrPending(ctx context.Context, t *testing.T, clientset ku
 	})
 
 	if err != nil {
-		if errors.Is(err, context.DeadlineExceeded) || strings.Contains(err.Error(), "timed out") || strings.Contains(err.Error(), "context deadline exceeded") {
+		if errors.Is(err, context.DeadlineExceeded) || strings.Contains(err.Error(), "timed out") || strings.Contains(err.Error(), "context deadline") {
 			t.Logf("Negative job %s successfully remained suspended or pending with 0 bound pods throughout the %v verification window.", jobName, verificationWindow)
 		} else {
 			t.Fatalf("Failed to verify negative job: %v", err)

